src/main.rs

```diff
@@ -135,7 +136,18 @@ fn main() -> Result<()> {
 
     let query_string = opts.read_query_to_string().transpose()?;
 
-    let input = BytesContainer::new(BytesContainerType::Input, opts.codec, buffer)?;
+    let engine = function_runner::engine::new_engine()?;
+    let module = Module::from_file(&engine, &opts.function)
+        .map_err(|e| anyhow!("Couldn't load the Function {:?}: {}", &opts.function, e))?;
+
+    // Infer codec from the module if it uses msgpack, otherwise use the codec from the command line
+    let inferred_codec = if function_runner::engine::uses_msgpack_provider(&module) {
+        Codec::Messagepack
+    } else {
+        opts.codec
+    };
```

Comment:
We're overriding the user-specified codec, which feels odd. Would it make more sense for the `codec` argument to be optional (no default) and then if it's `None`, we infer it?

Reply:
I feel like we might be able to remove the `codec` at this point? IIRC, we only added it for ourselves when we were experimenting with argo